File: backend/src/main/java/com/rivero/userAuth/service/AuthService.java
package com.rivero.userAuth.service;

import com.rivero.userAuth.model.User;
import com.rivero.userAuth.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
@Service
public class AuthService {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PasswordEncoder passwordEncoder;

    // Feature 1: Register User (CREATE)
    public User registerUser(User user) {
        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
            throw new RuntimeException("Username already exists");
        }
        // Hash the password before saving
        user.setPasswordHash(passwordEncoder.encode(user.getPasswordHash()));
        return userRepository.save(user);
    }

    // Feature 2: Login User (READ/VERIFY)
    public User loginUser(String username, String rawPassword) {
        User user = userRepository.findByUsername(username)
                .orElseThrow(() -> new RuntimeException("User not found"));

        if (!passwordEncoder.matches(rawPassword, user.getPasswordHash())) {
            throw new RuntimeException("Invalid credentials");
        }
        return user;
    }
}

File: mobile/app/src/main/java/com/rivero/userauth/api/AuthService.kt
package com.rivero.userauth.api

import com.rivero.userauth.model.User
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface AuthService {
    @POST("auth/register")
    fun register(@Body user: User): Call<User>

    @POST("auth/login")
    fun login(@Body user: User): Call<User>

    @POST("auth/logout")
    fun logout(): Call<Void>

    @GET("user/me")
    fun getProfile(): Call<User>
}